docs/sources.md

```diff
@@ -0,0 +1,246 @@
+Additionally contexts can be passed in from the docker cli: `docker build --build-context <name>=<path>`.
+The `<name>` would be the name to use in your yaml to access it.
+
+The `context` source type is considered to be a "directory" source.
+
+### Inline
+
+Inline sources are sources that are defined inline in the Dalec configuration.
+You can only specify one of `file` or `dir` in an inline source.
+Directories cannot be nested in inline sources.
+Filenames must not contain a path separator (`/`).
+
+```yaml
+sources:
+  someInlineFile:
+    inline:
+      # This is the content of the source.
+      file:
+        uid: 0
+        gid: 0
+        permissions: 0644
+        contents: |
+          some content
+          some more content
+
+  someInlineDir:
+    inline:
+      dir:
+        uid: 0
+        gid: 0
+        permissions: 0755
+        files:
+          # This is the content of the source.
+          file1:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+          file2:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+```
+
+Inline sources with `file` are considered to be "file" sources.
+Inline sources with `dir` are considered to be "directory" sources.
+
+### Docker Image
+
+Docker image sources fetch a docker image from a registry.
+The output of this source is a directory containing the contents of the image.
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+```
+
+You can also run commands in the image before fetching the contents.
+This is especially useful for generating content.
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+      cmd:
+        steps:
+          - command: echo ${FOO} ${BAR}
+            env:
+              FOO: foo
+              BAR: bar
+```
+
+You can mount any other source type into the image as well.
+Here's an example mounting an inline source:
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+      cmd:
+        mounts:
+          - dest: /foo
+            spec:
+              inline:
+                file:
+                  uid: 0
+                  gid: 0
+                  permissions: 0644
+                  contents: |
+                    some content
+                    some more content
+        steps:
+          - command: cat /foo
+```
+
+You can use the docker image source to produce any kind of content for your build.
+
+Docker image sources are considered to be "directory" sources.
+
+### Build
+
+Build sources allow you to build a dockerfile and use the resulting image as a source.
+It takes as input another source which must include the dockerfile to build.
+The default dockerfile path is `Dockerfile` just like a normal docker build.
```

Comment:
does this build source have a field to change the path to the dockerfile? In case these is not the default name `Dockerfile`?

Reply:
Added an example.